python/ray/serve/_private/proxy_state.py

```diff
@@ -387,111 +394,48 @@ def try_update_status(self, status: ProxyStatus):
 
-    def _health_check(self):
-        """Perform periodic health checks."""
-        assert self._status in {ProxyStatus.HEALTHY, ProxyStatus.DRAINING}
-
-        if self._actor_proxy_wrapper.health_check_ongoing:
-            try:
-                healthy_call_status = self._actor_proxy_wrapper.is_healthy()
-                if healthy_call_status == ProxyWrapperCallStatus.FINISHED_SUCCEED:
-                    # Call to reset _consecutive_health_check_failures
-                    # the status should be unchanged.
-                    self.try_update_status(self._status)
-                elif healthy_call_status == ProxyWrapperCallStatus.FINISHED_FAILED:
-                    self.try_update_status(ProxyStatus.UNHEALTHY)
-                elif (
-                    self._timer.time() - self._last_health_check_time
-                    > PROXY_HEALTH_CHECK_TIMEOUT_S
-                ):
-                    # Health check hasn't returned and the timeout is up, consider it
-                    # failed.
-                    self._actor_proxy_wrapper.reset_health_check()
-                    logger.warning(
-                        "Didn't receive health check response for proxy "
-                        f"{self._node_id} after {PROXY_HEALTH_CHECK_TIMEOUT_S}s"
-                    )
-                    self.try_update_status(ProxyStatus.UNHEALTHY)
-            except Exception as e:
-                logger.warning(f"Health check for proxy {self._actor_name} failed: {e}")
-                self.try_update_status(ProxyStatus.UNHEALTHY)
-
-        # If there's no active in-progress health check, and it has been more than 10
-        # seconds since the last health check, perform another health check.
-        if self._actor_proxy_wrapper.health_check_ongoing:
-            return
-        randomized_period_s = PROXY_HEALTH_CHECK_PERIOD_S * random.uniform(0.9, 1.1)
-        if self._timer.time() - self._last_health_check_time > randomized_period_s:
-            self._last_health_check_time = self._timer.time()
-            self._actor_proxy_wrapper.start_new_health_check()
-
-    def _drain_check(self):
-        """Check whether the proxy actor is drained or not."""
-        assert self._status == ProxyStatus.DRAINING
-
-        if self._actor_proxy_wrapper.is_draining:
-            try:
-                drained_call_status = self._actor_proxy_wrapper.is_drained()
-                if drained_call_status == ProxyWrapperCallStatus.FINISHED_SUCCEED:
-                    self.set_status(ProxyStatus.DRAINED)
-            except Exception as e:
-                logger.warning(f"Drain check for proxy {self._actor_name} failed: {e}.")
-        elif (
-            self._timer.time() - self._last_drain_check_time
-            > PROXY_DRAIN_CHECK_PERIOD_S
-        ):
-            self._last_drain_check_time = self._timer.time()
-            self._actor_proxy_wrapper.start_new_drained_check()
+    def reconcile(self, draining: bool = False):
```

Comment:
Let's leave this as `update` please (I agree `reconcile` is a better name but `update` is used throughout the code at the moment for similar patterns).

Also, in general please try to avoid unnecessary renaming/refactoring in P0 bug fixes, especially ones that will be cherry-picked, as they add risk and make it harder to review the changes.

Reply:
Agreed, i'm gonna be putting out a smaller, bare-bone PR just to patch the gap, and leave this one for 2.10